datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -262,7 +718,25 @@ async fn except_rels(
 
 /// Convert Substrait Plan to DataFusion LogicalPlan
 pub async fn from_substrait_plan(
-    state: &dyn SubstraitPlanningState,
+    state: &SessionState,
+    plan: &Plan,
+) -> Result<LogicalPlan> {
+    // Register function extension
+    let extensions = Extensions::try_from(&plan.extensions)?;
+    if !extensions.type_variations.is_empty() {
+        return not_impl_err!("Type variation extensions are not supported");
+    }
+
+    let consumer = DefaultSubstraitConsumer {
+        extensions: &extensions,
```

Comment:
I started thinking whether it actually  really make sense to pass the Extensions into the SubstraitConsumer, or if there should be some `initialize(plan: &Plan)` method or something in the SubstraitConsumer that loads them. But I didn't think yet much about it, maybe there is a reason to rather have it like this, and anyways no need to change anything for this PR anymore in any case 😄

Reply:
I do want to streamline the API further, but in the future 😅

Initialising the consumer directly from the plan makes sense to me, because we always need the extensions.